app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -34,22 +36,80 @@ interface SearchState extends ParsedQs {
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
+  const [cookies] = useCookies(["googtrans"]);
+  const history = useHistory();
+  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
-  const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
 
-  const searchState: SearchState = useMemo(
```

Comment:
I'm not exactly sure why useMemo was used here to begin with. Anyhow, `qs.parse...` was run each time there was any kind of editing of the filters in this previous setup, so AFAIK, there was no reason to memoize this value. For the new implementation, search updates, both to the actual search query and the filters, are handled by the searchState state variable and a useEffect block